napari/layers/points/points.py

```diff
@@ -1610,7 +1615,7 @@ def _update_draw(
         super()._update_draw(
             scale_factor, corner_pixels_displayed, shape_threshold
         )
-        self._set_highlight(force=True)
```

Comment:
Hm. Is this change not enough? There is a reason why someone use force here. Has this changed?

Reply:
This change is enough to break the cycle, but the update (with `force=True`) was added in #5737 and #5802 to fix the highlights when zooming. So this should be the same - forces an update if the scale factor is changed but not otherwise.

Reply:
maybe just
```
        prev_scale = self.scale_factor
        super()._update_draw(
            scale_factor, corner_pixels_displayed, shape_threshold
        )
        self._set_highlight(force=(prev_scale != scale_factor))
```
To avoid additional parameters?

Reply:
Ha - this was my initial proposal in https://github.com/napari/napari/issues/6223#issuecomment-1718175732.

I like it better as well, though 188438f perhaps matches more closely how other parameters are handled in `_set_highlight` so I could go either way. I'll update the PR and see if anyone else has feedback.